src/logic/shorten-url.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import fetch from 'node-fetch';
+
+import { config } from '../config';
+import { getLogger } from '../log';
+
+class BitlyConfigurationError extends Error {
+  name = 'BitlyConfigurationError';
+  status = 500;
+  expose = true; // The message will be exposed to users
+}
+
+export async function shortenUrl(longUrl: string): Promise<string> {
+  const log = getLogger('shortenUrl');
+  if (!config.bitlyToken) {
+    log.error('No access token for bitly has been configured!');
+    throw new BitlyConfigurationError(
+      'No access token for bitly has been configured!'
+    );
+  }
+
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/shorten';
+  const payload = { long_url: longUrl };
+
+  const response = await fetch(bitlyQueryUrl, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${config.bitlyToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
+
+  if (!response.ok) {
+    throw new Error(
+      `An error happened while shortening the long url ${longUrl}: ${response.statusText} (${response.status})`
+    );
+  }
+
+  const json = await response.json();
+  return json.link;
+}
+
+export async function expandUrl(urlToExpand: string): Promise<string> {
+  const log = getLogger('expandUrl');
+  if (!config.bitlyToken) {
+    log.error('No access token for bitly has been configured!');
+    throw new BitlyConfigurationError(
+      'No access token for bitly has been configured!'
+    );
+  }
+
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/expand';
+  const payload = {
+    bitlink_id: urlToExpand.replace(/^https:\/\//, ''),
```

Comment:
It's also not clear to me from the docs what is a valid string here: https://dev.bitly.com/v4/#operation/expandBitlink Do we need to remove the https? Can this be abused by providing domains that we don't support? Should we add an allow-list of domains? Is that over-engineering?

Reply:
This was found by trial and error, the v3 API was easier for this :/

bitlink_id is actually returned by the 'shorten' API.

My approach here is that because we return an error if it's not a profiler URL, abusing this isn't very useful.
Also, we have to support bit.ly because some of our URLs use it :(